peps/pep-0396.rst

```diff
@@ -25,14 +31,25 @@ Conformance with this PEP is optional, however other Python tools
 (such as ``distutils2`` [1]_) may be adapted to use the conventions
 defined here.
 
-PEP Rejection
-=============
+
+PEP Rejection/Withdrawal
+========================
 
 This PEP was formally rejected on 2021-04-14.  The packaging ecosystem
 has changed significantly in the intervening years since this PEP was
 first written, and APIs such as ``importlib.metadata.version()`` [11]_
 provide for a much better experience.
 
+This rejection was subsequently reclassified as a withdrawal,
+as the previous state was being misinterpreted [12]_ as suggesting
+that *no* modules should be defining ``__version__`` attributes,
+which definitely isn't the case.
+
+Modules may still define ``__version__`` if they choose to,
```

Comment:
Suggest not saying the "choose not to do so" part.  Given that many build backends can dynamically refer to static `__version__` in a file and use that to influence the built package's version scheme (and thus the `importlib.metadata.version()` value), it can still make sense to set it, under the principle of Single Source of Truth.  Of course there are many ways to handle this, so it's completely within the prerogative of the package authors.

I'll also suggest mentioning this type of dynamic version number use in backends like hatchling and such.

Reply:
I do want to say that leaving it out doesn't break anything fundamental, since that's the main reason this convention doesn't need to be a PEP.

Mentioning tooling support is a good idea though, so I'll reword this to include links to the user guide sections and borrow some of their wording.

Reply:
I went with the following wording:

```
Modules are still free to define ``__version__`` if they choose to.
However, choosing *not* to do so won't interfere with looking up
the version information for installed distribution packages, so an
Informational PEP isn't the right tool to document community
conventions around the use of module ``__version__`` attributes
(they're better covered as part of the Python Packaging User Guide).
```